src/dune_rules/merlin.ml

```diff
@@ -61,38 +60,43 @@ module Processed = struct
           [ make_directive "FLG" (Sexp.List (List.map ~f:to_atom flags)) ]
       in
       match pp with
+      | None -> flags
       | Some (pp_flag, pp_args) ->
         make_directive "FLG" (Sexp.List [ Atom pp_flag; Atom pp_args ]) :: flags
-      | None -> flags
     in
     let suffixes =
-      Extensions.Set.to_list extensions
-      |> List.map ~f:(fun (impl, intf) ->
-             make_directive "SUFFIX"
-               (Sexp.Atom (Printf.sprintf "%s %s" impl intf)))
+      List.map extensions ~f:(fun { Ml_kind.Dict.impl; intf } ->
+          make_directive "SUFFIX" (Sexp.Atom (Printf.sprintf "%s %s" impl intf)))
     in
     Sexp.List
       (List.concat [ exclude_query_dir; obj_dirs; src_dirs; flags; suffixes ])
 
-  let rec get config ~filename =
-    let file = Filename.remove_extension filename in
-    match Option.map (String.Map.find config filename) ~f:to_sexp with
-    | Some result -> Some result
-    | None when String.equal file filename -> None
-    | None -> get config ~filename:file
+  let get { modules; pp_config; config } ~filename =
+    let fname = Filename.remove_extension filename in
```

Comment:
I think this `fname` should probably be lower cased as well. Or, you could  just take compare it to one of the `Module.source` of the module.